dask/array/core.py

```diff
@@ -3526,7 +3526,9 @@ def from_array(
     if meta is None:
         meta = x
 
-    return Array(dsk, name, chunks, meta=meta, dtype=getattr(x, "dtype", None))
+    arr = Array(dsk, name, chunks, meta=meta, dtype=getattr(x, "dtype", None))
+    arr_new = arr.to_backend(config.get("array.backend"))
```

Comment:
This is not going to work at all...

Reply:
The default argument for `to_backend` will be the current `"array.backend"` configuration. So, the following should do the same:

```python
arr_new = arr.to_backend()
```

With that in mind, the original backend-dispatching changes skipped over `from_array`, because (1) the code would need to be moved out of `dask/array/core.py` and into a dedicated io module, and (2) it doesn't necessarily make sense to use a backend configuration for **all** input arrays. That is, if we are reading from storage, then backend-dispatching makes sense. If we are creating an array from numpy/cupy, then the existing in-memory data should dictate the type of collection we create.

Reply:
>That is, if we are reading from storage, then backend-dispatching makes sense. If we are creating an array from numpy/cupy, then the existing in-memory data should dictate the type of collection we create.

For me personally, reading from storage with a specific backend is a much more important feature than creating a Dask array from numpy/cupy in-memory data.

Can these two tasks be separated? I haven't looked at this part of the code properly for a while, but I'm guessing everything has to go through the same path and we can't avoid the problem for now?

Reply:
>Can these two tasks be separated? I haven't looked at this part of the code properly for a while, but I'm guessing everything has to go through the same path and we can't avoid the problem for now?

I think you have this right. The two cases probably can (and should) be separated.  The danger of simply flipping a switch in `from_array` (for now) is that you will be affecting cases where the source is zarr/hdf5/netcdf **and**  the case that the collection is being created from existing cupy/numpy data.